core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorSidebarDirective.js

```diff
@@ -40,6 +41,17 @@ oppia.directive('simpleEditorSidebar', [
           $scope.questionList = SimpleEditorManagerService.getQuestionList();
           $scope.ID_PREFIX = QuestionIdService.SIDEBAR_PREFIX;
           $scope.sidebarModeService = SimpleEditorSidebarModeService;
+          $scope.addNewQuestion = function() {
+            if (!SimpleEditorManagerService.canAddNewQuestion()) {
+              return;
+            }
+            $scope.sidebarModeService.setModeToReadonly();
```

Comment:
@saeedjassani  and @seanlip  I rechecked the behaviour and the code. We are not expanding the questions ourselves, only those questions expanded before switching mode gets expanded i.e. the state perfectly restores what It was before changing the mode and it is the desired behaviour. The issue is mainly because the sidebar is not scrolling down along with the editor body. So , I think everything is great, we just need to scroll the sidebar too.